File: Arrays/5. Find the duplicate element in a limited range array/c++/code.cpp
#include <iostream>
#include <algorithm>
#include <vector>
#include <sstream>

using namespace std;

void printArray(vector<int> arr)
{
    for (int i = 0; i < arr.size(); i++)
    {
        cout << arr[i] << " ";
    }

    cout << endl;
}

int main()
{
    string input, str;
    getline(cin, input);
    stringstream ss(input);
    vector<int> arr;
    while (getline(ss, str, ' '))
    {
        arr.push_back(stoi(str));
    }

    vector<int> temp({arr[0]});

    for (int i = 0; i < arr.size(); i++)
    {
        auto itr = std::find(temp.begin(), temp.end(), arr[i]);
        if (itr == temp.end())
        {
            temp.push_back(arr[i]);
        }
    }

    printArray(temp);

    return 0;
}

// Input example - 1 2 3 4 2 2 3 4 1

File: Arrays/5. Find the duplicate element in a limited range array/Java/src/Code.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;
import java.util.StringTokenizer;

public class Code {
    public static void printArray(List<Integer> arr) {
        for (int i = 0; i < arr.size(); i++) {
            System.out.print(arr.get(i) + " ");
        }
        System.out.println("\n");
    }
    public static void main(String[] args) throws Exception {
        Scanner sc = new Scanner(System.in);
        String input = sc.nextLine();

        StringTokenizer stringTokenizer = new StringTokenizer(input);
        int counts = stringTokenizer.countTokens();

        List<Integer> arr = new ArrayList<>(counts);

        for (int i = 0; i < counts; i++) {
            arr.add(Integer.parseInt((String) stringTokenizer.nextElement()));
        }

        List<Integer> temp = new ArrayList<>();

        for (int i = 0; i < arr.size(); i++) {
            if (!temp.contains(arr.get(i))) {
                temp.add(arr.get(i));
            }
        }

        printArray(temp);

        sc.close();
    }
}

// Input example - 1 2 3 4 2 2 3 4 1
